packages/routing-utils/src/superstatic.ts

```diff
@@ -174,28 +182,60 @@ export function sourceToRegex(
   return { src: r.source, segments };
 }
 
+function collectHasSegments(has?: HasField) {
+  const hasSegments = new Set<string>();
+
+  for (const hasItem of has || []) {
+    if (!hasItem.value && (hasItem as any).key) {
+      hasSegments.add((hasItem as any).key);
+    }
+
+    if (hasItem.value) {
+      const matcher = new RegExp(`^${hasItem.value}$`);
+      const matches = matcher.exec('');
+
+      if (matches) {
+        if (matches.groups) {
+          Object.keys(matches.groups).forEach(groupKey => {
+            hasSegments.add(groupKey);
+          });
+        } else {
+          hasSegments.add((hasItem as any).key || 'host');
+        }
+      }
+    }
+  }
+  return [...hasSegments];
+}
+
 function replaceSegments(
   segments: string[],
+  hasItemSegments: string[],
   destination: string,
   isRedirect?: boolean
 ): string {
   const parsedDestination = parseUrl(destination, true);
-  delete parsedDestination.href;
-  delete parsedDestination.path;
-  delete parsedDestination.search;
+  delete (parsedDestination as any).href;
+  delete (parsedDestination as any).path;
+  delete (parsedDestination as any).search;
```

Comment:
What changed here 🤔

Reply:
This was showing an error since we're using delete on non-optional fields so casted it to any to remove.
 
<img width="1326" alt="Screen Shot 2021-04-01 at 7 10 19 PM" src="https://user-images.githubusercontent.com/22380829/113366147-fc3b2000-931d-11eb-9536-ac5b74fb472a.png">

Reply:
Oh, I'm surprised the build didn't fail before 🤔